### PR TITLE
libqedr: add support for SRQ's

### PR DESCRIPTION
**Pull request by yuvalbason**
Enable SRQ's for rdma-core.

SRQ rdma-core implementation is pretty straight forward, except perhaps the
Way that the driver udpates FW producers: they are updated using a mapped buffer
that FW reads, and not doorbells Like RQ / SQ.

Signed-off-by: Yuval Bason <yuval.bason@cavium.com>
Signed-off-by: Michal Kalderon <michal.kalderon@cavium.com>

### CODE REVIEW

**Comment by jgunthorpe** [comment]
You need to fix the build.

File: providers/qedr/qelr_verbs.c [hunk 1]
```diff
@@ -333,6 +508,11 @@ static void qelr_chain_free_rq(struct qelr_qp *qp)
 	qelr_chain_free(&qp->rq.chain);
 }
 
+static inline bool qelr_qp_has_srq(struct qelr_qp *qp)
+{
+	return !!qp->srq;
```
Comment (jgunthorpe) [inline providers/qedr/qelr_verbs.c]: !! obfuscation isn't needed if casting to bool

Comment (yuvalbason) [reply providers/qedr/qelr_verbs.c]: will fix.

**Comment by jgunthorpe** [comment]
Updated with the final kernel header commit

**Pull request merged**